Cargo.toml

```diff
@@ -1,18 +1,22 @@
 [package]
 name = "panel-firmware"
-version = "0.3.0"
+version = "0.4.0"
 authors = ["Brian Schwind <brianmschwind@gmail.com>"]
 license = "MIT"
 edition = "2018"
 
+[features]
+default = ["fan_control"]
+fan_control = []
```

Comment:
How we'd use the feature flag? I guess we'll build one firmware variant with the flag on, one variant with the flag off, and then make sure we flash correct firmware to correct devices. That would work.

But it would be more practical to build all non-outdated variants at once, and name them appropriately right in Cargo.toml (rather adding some docs with conventions): instead of the feature flag, we could have multiple "binaries", each for a given hardware variant. They could/should share almost all the code - the `main.rs` files could be very thin shims.

CC @bschwind who has the most context.

Reply:
Yes, if we have feature flags at all I'd prefer them to be based on board variants.

Currently our firmware only reports a short SHA1 hash for which commit it was built from, but no way to report what features it was built with.

We can do as you suggest and build multiple binaries, one for each hardware variant, or we could put more work into making the protocol flexible in handling boards of varying capabilities.

Either way, it's probably best to wait until we have the next hardware revision before deciding exactly what we do.